solidity/test/helpers/Contracts.js

```diff
@@ -0,0 +1,96 @@
+const { ethers } = require('hardhat');
+
+const contracts = {};
+
+const deployContract = async (contractName, ...args) => {
+    let signer = (await ethers.getSigners())[0];
+
+    if (typeof args[args.length - 1] === 'object' && args[args.length - 1].from) {
+        signer = args[args.length - 1].from;
+        args.pop();
+    }
+
+    if (contracts[contractName + signer.address] === undefined) {
```

Comment:
`contractName + signer.address` can be refactored to a single var so that it won't repeat itself